packages/request-manager/src/torIdentities.ts

```diff
@@ -0,0 +1,91 @@
+import http from 'http';
+import https from 'https';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { EventEmitter } from 'events';
+import { RequestInit } from 'node-fetch';
+
+interface TorConnectionOptions {
+    host: string;
+    port: number;
+}
+
+interface TorRequestOptions {
+    identity?: string;
+}
+
+export class TorIdentities extends EventEmitter {
+    options: TorConnectionOptions;
+    identities: { [key: string]: SocksProxyAgent };
+
+    constructor(options: TorConnectionOptions) {
```

Comment:
im wondering if this class shoudnt be done as singletone with static methods
currently if i would like to use it in different files/places in code i need either store class instance somewhere and pass it thru multiple layers or create new instance each time i want to use it.



maybe it would be better to seup it once, when tor is enabled:
```
const torController = new TorController({ .... });
TorIdentities.init(torController);


// and then use it somewhere else:

const r = await TorIdentities.request(url, { method: 'GET' }, { indentity: 'A' });
```

Reply:
Yes, that might be interesting given the use-case you are describing, what about 2e513bb0f2999bf63e5ad54444bd44817bd40a3d and 011496626d06b34e6b24a39118dd76164e4c655e  ?